app/middleware/middleware_healthcheck.rb

```diff
@@ -0,0 +1,32 @@
+class MiddlewareHealthcheck
+  def initialize(app)
+    @app = app
+  end
+
+  def call(env)
+    if env['PATH_INFO'.freeze] == '/status'.freeze
+      response = [200, { 'Content-Type' => 'application/json' }]
+
+      # checks the database health
+      begin
+        db_healthy = ActiveRecord::Migrator.current_version != 0
+      rescue StandardError => e
+        Rails.logger.info "Database error: #{e}"
+        db_healthy = false
+      end
+
+      # checks the health of launchpad
+      cmr_client = Cmr::Client.client_for_environment(Rails.configuration.cmr_env, Rails.configuration.services)
+      launchpad_healthy = cmr_client.launchpad_healthcheck.body == 'OK'.freeze
```

Comment:
So when I was testing this call previously, if we were unable to reach launchpad the call would fail pretty quickly. I'm not sure what is different now, but locally, if I am not on the vpn, this call hangs for a while and the page we are expecting (showing the response as text) doesn't show. I am wondering if we should add a timeout so if the call to launchpad doesn't finish within that time it will allow the response to be rendered. Since it is happening locally, it is possible it could happen in our other environments also.
What do you think, @jceaser ?

Reply:
yes, I think we should, the difference might be in how it fails, if the client is waiting for a DNS resolution vs the actual server. The VPN may affect this. Lets try specifying a timeout, I think if we don't hear from launchpad in a second, we have a problem.